frontend/src/search/SearchBar.tsx

```diff
@@ -0,0 +1,34 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useGlobalContext } from "../auth/GlobalContext";
+
+const SearchBar: React.FC = () => {
+  const { refresh, setRefresh } = useGlobalContext();
+  const [search, setSearch] = useState("");
+  const navigate = useNavigate();
+
+  const searchHandler = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (!search) return;
```

Comment:
I'm quite sure search is never null. Maybe change to `if (!search || search.length === 0)`